multiDist/GoldRush.js

```diff
@@ -0,0 +1,104 @@
+class GoldRush extends Matrix {
+    constructor() {
+        super()
+        this.coins = 0
+        this.matrix = []
+        this.player1 = {
+            position: [0, 0],
+            score: 0
+        }
+        this.player2 = {
+            position: [],
+            score: 0
+        }
+        this.win = 0
+    }
+
+    generateBoard(numRows, numColumns) {
+        this.coins = 0
+        let matrix = []
+        this.player2.position = [numRows - 1, numColumns - 1]
+        for (let r = 0; r < numRows; r++) {
+            matrix.push([])
+            for (let c = 0; c < numColumns; c++) {
+                if(r === 0 && c === 0) { matrix[r].push(1) }
+                else if (r === numRows - 1 && c === numColumns - 1) { matrix[r].push(2) }
+                else {
+                    let chance = Math.floor(Math.random() * 10)
+                    if (chance < 6){
+                        matrix[r].push('e')
+                    } else if (chance < 8) {
+                        matrix[r].push('c')
+                        this.coins += 1
+                    } else {
+                        if (r > 0 && (matrix[r - 1][c - 1] === 'w' || matrix[r - 1][c + 1] === 'w')) {
+                            matrix[r].push('e')
+                        } else {
+                            matrix[r].push('w')
+                        }
+                    }
+                }
+            }
+        }
+        return matrix
+    }
```

Comment:
you could deconstruct this method into little methods - each `if` statement could be a method and that way it would be more readable and understandable (clean code)